model/inline/src/main/scala/aqua/model/inline/raw/MakeAbilityRawInliner.scala

```diff
@@ -1,31 +1,39 @@
 package aqua.model.inline.raw
 
-import aqua.model.{
-  CallModel,
-  CallServiceModel,
-  LiteralModel,
-  OpModel,
-  SeqModel,
-  ValueModel,
-  VarModel
-}
-import aqua.model.inline.raw.RawInliner
-import cats.data.Chain
-import aqua.model.inline.state.{Arrows, Exports, Mangler}
-import aqua.raw.value.{AbilityRaw, LiteralRaw, MakeStructRaw}
-import cats.data.{NonEmptyList, NonEmptyMap, State}
 import aqua.model.inline.Inline
-import aqua.model.inline.RawValueInliner.{unfold, valueToModel}
-import aqua.types.{ArrowType, ScalarType}
-import cats.syntax.traverse.*
-import cats.syntax.monoid.*
-import cats.syntax.functor.*
-import cats.syntax.flatMap.*
-import cats.syntax.apply.*
+import aqua.model.inline.RawValueInliner.unfold
+import aqua.model.inline.state.{Arrows, Exports, Mangler}
+import aqua.model.{SeqModel, ValueModel, VarModel}
+import aqua.raw.value.AbilityRaw
+import aqua.types.AbilityType
+import cats.Eval
+import cats.data.{Chain, IndexedStateT, NonEmptyMap, State}
 import cats.syntax.foldable.*
 
 object MakeAbilityRawInliner extends RawInliner[AbilityRaw] {
 
+  private def updateFields[S: Mangler: Exports: Arrows](
+    name: String,
+    fields: NonEmptyMap[String, (ValueModel, Inline)]
+  ): State[S, Unit] = {
+    for {
+      res <- fields.toNel.traverse {
+        case (n, (vm@VarModel(vName, at@AbilityType(_, _), properties), _)) =>
+          val leftName = AbilityType.fullName(name, n)
+          for {
+            _ <- Exports[S].renameAbilityPrefix(vName, leftName)
```

Comment:
Why rename?

I believe, because of this such code does not work:
```
closure = (x: i8) -> bool:
    <- x > 0

mySimple = Simple(
    arrow = closure
)

myComplex = Complex(
    simple = mySimple,
    arrow = mySimple.arrow,
)

mySimple.arrow(42), myComplex.arrow(42)
```

should not it work?

Reply:
you are right. Will fix this